src/NuGet.Core/NuGet.Commands/Utility/MSBuildProjectFrameworkUtility.cs

```diff
@@ -221,16 +298,37 @@ public static NuGetFramework GetProjectFrameworkReplacement(NuGetFramework frame
             return framework;
         }
 
-        private static string GetPropertyOrNull(string propertyName, IDictionary<string, string> projectProperties)
+        /// <summary>
+        /// Get a NuGetFramework out of the passed object. The argument is expected to either be a <see cref="NuGetFramework"/> or <see cref="string"/>.
+        /// </summary>
+        private static NuGetFramework GetAsNuGetFramework(object arg)
+        {
+            if (arg is NuGetFramework)
+            {
+                return (NuGetFramework)arg;
+            }
+            if (arg is string frameworkString)
+            {
+                return NuGetFramework.Parse(frameworkString);
+            }
+            throw new ArgumentException("Unexpected object type");
+        }
+
+        /// <summary>
+        /// Get a roundtrippable framework string out of the passed object. The argument is expected to either be a <see cref="NuGetFramework"/> or <see cref="string"/>.
+        /// </summary>
+        private static string GetAsFrameworkString(object arg)
```

Comment:
ditto here. I feel like this is Too Clever™, and I don't like bypassing static type systems?

Reply:
My reasoning is in: https://github.com/NuGet/NuGet.Client/pull/3562#discussion_r465908574